### PR TITLE
Update Inno Setup Log file after update

### PR DESCRIPTION
**Pull request by neon-nyan**
# What's changed?
This PR includes a change for updating the "Inno Setup Log"'s ``unins000.dat`` file, which is a manifest to list all the files to be removed on uninstalling the app.

Currently due to the nature of Collapse using Squirrel as its Update module and Inno Setup as its Installer for both Stable and Preview builds, all the updated files related with Collapse doesn't get registered in the ``unins000.dat`` file and making the uninstallation incomplete.

This PR also adding another submodule [**InnoSetupLogParser**](https://github.com/CollapseLauncher/InnoSetupLogParser/tree/main), which is a parser and modification tool for ``unins000.dat`` file. This submodule was initially created by [**preseverence**](https://github.com/preseverence/isulr) and has been modified to work on both Collapse and [**ApplyUpdate**](https://github.com/CollapseLauncher/ApplyUpdate).

This PR is to solve the issue mentioned in #345.

### CODE REVIEW

File: CollapseLauncher/XAMLs/MainApp/MainPage.xaml.cs [hunk 1]
```diff
@@ -641,11 +643,24 @@ private void SpawnAppUpdatedNotification()
...
+                // If the update was handled by squirrel and it needs Inno Setup Log file to get updated, then do the routine
+                if (File.Exists(NeedInnoUpdateFile))
+                {
+                    try
+                    {
+                        string InnoLogPath = Path.Combine(Path.GetDirectoryName(AppFolder), "unins000.dat");
+                        if (File.Exists(InnoLogPath)) InnoSetupLogUpdate.UpdateInnoSetupLog(InnoLogPath);
+                        File.Delete(NeedInnoUpdateFile);
+                    }
+                    catch { }
```
Comment (Cryotechnic) [inline CollapseLauncher/XAMLs/MainApp/MainPage.xaml.cs:661]: Might want to do something in the catch to log the error in the event that it happens.

File: InnoSetupHelper/InnoSetupLogUpdate.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            "config.ini",
+            "_Temp",
+            "unins00",
+            "unins000",
+
+            // Game Files (we added this in-case the user uses the same directory as the executable)
+            "Hi3SEA", "Hi3Global", "Hi3TW", "Hi3KR", "Hi3CN", "Hi3JP", "BH3",
+            "GIGlb", "GICN", "GenshinImpact", "YuanShen", "GIBilibili",
+            "SRGlb", "SRCN", "StarRail", "HSRBilibili",
+            "ZZZGlb", "ZZZCN", "ZZZ"
```
Comment (Cryotechnic) [inline InnoSetupHelper/InnoSetupLogUpdate.cs:32]: ZZZ will probably have Bilibili. Also, how do we handle this in the event that the user creates a custom folder name which falls outside of these ones? Maybe we would want to grab this from the config in Collapse directly?

**Review by Cryotechnic — CHANGES_REQUESTED** [review]
Just small changes but aside from that looks good!

File: InnoSetupHelper/InnoSetupLogUpdate.cs (hunk 2 above)
Comment (bagusnl) [reply InnoSetupHelper/InnoSetupLogUpdate.cs:32]: I believe we already have a check that warns people to not use the same folder as collapse main app does to save game files, so I think it shouldn't be necessary

**Review by bagusnl — COMMENTED** [review]
Looks all fine for me, just a question. But before approval, I just need to make sure that this accounts for forceUpdate and branch switching

**Comment by Cryotechnic** [comment]
Approved

 Yes

**Review by Cryotechnic — APPROVED** [review]

**Pull request merged**